### PR TITLE
CAPI1010 addressing Issue50

### PR DESCRIPTION
**Pull request by DaemonWill**
### Requirements

Upon login and entering the services page, admin is able to see all of the services in the database regardless of which user created the service. For this scenario, only grant the admin access to the data for the services created by them, and not by other users. 

### Description of the Change

Current fix applies a db search filter for services created by the current user if the original http method of the request was a "GET," this includes admin users. Admin maintenance of the app/services for updating, creating, and deleting should still be in place unless they are also called by "GET."

### Benefits

TODO

### Concerns

Could affect Jenkins builds concerning copying, deleting, or updating existing services.

### CODE REVIEW

File: platform_services/components/crud/getList.js [hunk 1]
```diff
@@ -22,10 +22,10 @@
 	@version: 1.0
 **/
 
-const utils = require("../utils.js")(); 
+const utils = require("../utils.js")();
 const _ = require("lodash");
 
-module.exports = (query, onComplete) => {
+module.exports = (query, method, onComplete) => {
```
[the remark this answers is not in the document]
Comment (DaemonWill) [reply platform_services/components/crud/getList.js]: That's doable; however, I'll be assigning that boolean not only on the user's status as an admin, but also based off of the method used. We currently have logic to check for the user's admin status and filtering records accordingly so I'm thinking this additional criteria for the boolean is needed, my only concern is if there are any jenkins builds that will require an admin to do crud.getList to retrieve all the services before an update/delete

[the remark this answers is not in the document]
Comment (DaemonWill) [reply platform_services/components/crud/getList.js]: I'll make the change first and update the comment accordingly to addresses the potential concern.

[the remark this answers is not in the document]
Comment (DaemonWill) [reply platform_services/components/crud/getList.js]: parameter and logic is in place, the additional criteria being checked for is the event.method on top of admin status. Does this seem like a valid approach?

File: platform_services/components/crud/getList.js [hunk 2]
```diff
@@ -80,11 +79,11 @@ module.exports = (query, onComplete) => {
 				attributeValues[(":" + key_name)] = {
 					'S': query[key]
 				};
-			}   
+			}
 		});
     }
 
-    if (global.userId && !_.includes(global.config.admin_users, global.userId.toLowerCase())) {
```
Comment (suryajak) [inline platform_services/components/crud/getList.js]: This logic should reside in the crud only. The consumer of the crud can ask for all records, but if they are not the admin(s), it should just return the current logged in user records.

File: platform_services/index.js [hunk 3]
```diff
@@ -320,7 +332,8 @@ module.exports.handler = (event, context, cb) => {
                 },
                 // Check if a service with same domain and service_name combination exists
                 validateServiceExists: function(onComplete) {
-                    crud.getList({ 'service': service_data.service, 'domain': service_data.domain }, function onServiceGet(error, data) {
+                    getAllRecords = getAllCheck(event.method);
+                    crud.getList({ 'service': service_data.service, 'domain': service_data.domain }, getAllRecords, function onServiceGet(error, data) {
```
Comment (suryajak) [inline platform_services/index.js]: Here it should say: crud.getList({ 'service': service_data.service, 'domain': service_data.domain }, getAllRecords: true,

File: platform_services/components/crud/getList.js (hunk 2 above)
Comment (DaemonWill) [reply platform_services/components/crud/getList.js]: Ok that makes sense. Thanks for the clarification

File: platform_services/components/crud/getList.js [hunk 4]
```diff
@@ -80,11 +80,11 @@ module.exports = (query, onComplete) => {
...
+    if ((global.userId && !_.includes(global.config.admin_users, global.userId.toLowerCase())) || !getAllRecords) {
```
Comment (suryajak) [inline platform_services/components/crud/getList.js]: small optimization based on how OR works [short circuit eval](https://en.wikipedia.org/wiki/Short-circuit_evaluation) - you want to move !getAllRecords to the front

**Pull request merged**